src/Core/src/Platform/Windows/DatePickerExtensions.cs

```diff
@@ -38,9 +38,31 @@ public static void UpdateMaximumDate(this CalendarDatePicker platformDatePicker,
 
 		public static void UpdateCharacterSpacing(this CalendarDatePicker platformDatePicker, IDatePicker datePicker)
 		{
-			platformDatePicker.CharacterSpacing = datePicker.CharacterSpacing.ToEm();
+			// Store the character spacing value to apply it when ready
+			var characterSpacing = datePicker.CharacterSpacing;
+
+			// Apply immediately if loaded, otherwise wait for load
+			if (platformDatePicker.IsLoaded)
+			{
+				ApplyCharacterSpacingToTextBlocks(platformDatePicker, characterSpacing);
+			}
+			else
+			{
+				// Wait for the control to load, then apply character spacing
+				platformDatePicker.OnLoaded(() => ApplyCharacterSpacingToTextBlocks(platformDatePicker, characterSpacing));
```

Comment:
Could be a potential Memory Leak. For example:
- If UpdateCharacterSpacing is called multiple times before the control loads, multiple event handlers get registered
- There's no cleanup mechanism if the control is disposed before loading


```
platformDatePicker.Loaded -= OnDatePickerLoaded;
  
  void OnDatePickerLoaded(object sender, RoutedEventArgs e)
  {
      if (sender is CalendarDatePicker picker)
      {
          // Clean up handler immediately
          picker.Loaded -= OnDatePickerLoaded;
          ApplyCharacterSpacingToTextBlocks(picker, characterSpacing);
      }
  }
  
  platformDatePicker.Loaded += OnDatePickerLoaded;
```

Reply:
@jsuarezruiz , Based on suggestion, I've modified the fix